aptos-move/framework/aptos-stdlib/sources/cmp.move

```diff
@@ -0,0 +1,90 @@
+module std::cmp {
+    const EQUAL: u8 = 1;
+    const LESS_THAN: u8 = 0;
+    const GREATER_THAN: u8 = 2;
+
+    /// As there are no signed values in move, all values are shifted by 1 up.
+    /// An int value:
+    /// 1   iff both values are the same
+    /// 0   iff first value is smaller than the second
+    /// 2   iff first value is larger than the second
+    native fun compare_impl<T>(first: &T, second: &T): u8;
+
+    struct Ordering has copy, drop {
```

Comment:
Can we use enum for this now? cc @wrwg

Reply:
would enum be better here? Would enums be more inefficient here? 

how would the native function return an enum that is defined in the module here?